src/store/reducers/nodesList.ts

```diff
@@ -36,13 +44,17 @@ const nodesList = function (state = initialState, action) {
 
 export function getNodesList() {
     return createApiRequest({
-        request: window.api.getNodeInfo(),
+        request: window.api.getNodesList(),
         actions: FETCH_NODES_LIST,
-        dataHandler: (data) => {
-            const {SystemStateInfo: nodes = []} = data;
-            return nodes;
-        },
     });
 }
 
+export const selectNodesMap = (state: NodesListRootStateSlice) =>
+    state.nodesList.data?.reduce<NodesMap>((nodesMap, node) => {
+        if (node.Id && node.Host) {
+            nodesMap[node.Id] = node.Host;
+        }
+        return nodesMap;
+    }, {});
```

Comment:
Let's use Map here, it is more efficient, when it is needed to change values inside.